ringbuffer.cpp

```diff
@@ -0,0 +1,152 @@
+#include <unistd.h>
+
+#include "ringbuffer.h"
+
+#include "log.h"
+
+template<typename T>
+Ringbuffer<T>::Ringbuffer(size_t num_elem)
+{
+    _buffer = (uint8_t *)calloc(num_elem, sizeof(T));
+    _num_elem = num_elem;
+}
+
+template<typename T>
+Ringbuffer<T>::~Ringbuffer()
+{
+    free(_buffer);
+}
+
+template<typename T>
+enum Ringbuffer<T>::add_result Ringbuffer<T>::add(T *item, uint32_t seqno)
```

Comment:
This is not a regular Ringbuffer implementation it is specialized for BinLog usage so maybe is better have a more specific name.